tests/ClientFunctionalTest.php

```diff
@@ -137,4 +141,25 @@ public function testAddAndRemoveSubscriber(): void
 
         $client->getManager()->executeCommand('admin', new Command(['ping' => 1]));
     }
+
+    public function testWatchWithBuilderPipeline(): void
+    {
+        $this->skipIfChangeStreamIsNotSupported();
+
+        if ($this->isShardedCluster()) {
+            $this->markTestSkipped('Test does not apply on sharded clusters: need more than a single getMore call on the change stream.');
+        }
```

Comment:
I assume this was copied from `testChangeStreamExample_1_4` in DocumentationExamplesTest. I think this skip was only done for documentation examples to avoid complicating the code snippets.

https://github.com/mongodb/mongo-php-library/commit/32bade4780e432251e448a34ccf39b39baea0e33 also introduced [`advanceCursorUntilValid()`](https://github.com/mongodb/mongo-php-library/commit/32bade4780e432251e448a34ccf39b39baea0e33#diff-e684dc1dab05b372a1d383b95d0fde73fdc3a79ddc8559aeda884d644b3d2caeR1673) to fix change stream iteration for sharded clusters in our other tests.

I won't push for changing this, but wanted to provide some context.

Reply:
Thanks for the context. In fact, I've simplified the documentation example you pointed. Given what's being tested, I don't think it's essential for the test to run on all server typologies.